cyclops/monitor/clinical_applicator.py

```diff
@@ -19,182 +18,190 @@ class ClinicalShiftApplicator:
 
     """
 
-    def __init__(self, shift_type: str, source, target):
+    def __init__(self, shift_type: str, source, target, shift_id: str = None):
         self.shift_type = shift_type
+        self.shift_id = shift_id
+        self.source = source
+        self.target = target
 
-        self.method_args = {"source": source, "target": target}
-
-        self.shift_types = {
+        self.shift_types: Dict[str, Callable[..., Dataset]] = {
             "time": self.time,
             "month": self.month,
             "hospital_type": self.hospital_type,
+            "custom": self.custom,
```

Comment:
I don't see a `custom` attribute

Reply:
It's points to the method within the class called "custom" at the bottom of the script.